src/components/auth/UserAuthForm.tsx

```diff
@@ -0,0 +1,119 @@
+// src/components/auth/UserAuthForm.tsx
+import React, { useState } from 'react';
+import { useUserAuth } from '../../lib/UserAuthContext';
+
+export function UserAuthForm({ onSuccess }: { onSuccess?: () => void }) {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
```

Comment:
Esto podemos ponerlo como
```suggestion
  const [userForm, setUserForm] = useState({
      email: '',
      password: ''
  })
```